src/foam/util/AddressUtil.java

```diff
@@ -112,28 +112,21 @@ public static String normalizeCountry(X x, String countryCode) {
       return countryCode;
     }
     
-    String[] normalizedCountry = { countryCode };
+    String normalizedCountry = countryCode;
     DAO countryDAO = (DAO) x.get("countryDAO");
-    Country found = (Country) countryDAO.find(countryCode);
-    if ( found != null )
-      return found.getCode();
+    Country country = (Country) countryDAO.find(countryCode);
+    if ( country != null )
+      return country.getCode();
 
-    countryDAO
-      .where(OR(
-        EQ(Country.ISO31661CODE, countryCode),
-        STARTS_WITH_IC(Country.NAME, countryCode)
-      ))
-      .select(new AbstractSink() {
-        public void put(Object o, Detachable d) {
-          Country country = (Country) o;
-          if ( country.getIso31661Code().equals(countryCode) || country.getName().equalsIgnoreCase(countryCode) ) {
-            normalizedCountry[0] = country.getCode();
-            d.detach();
-          }
-        }
-      });
+    country = (Country) countryDAO.find(OR(
+      EQ(Country.ISO31661CODE, countryCode),
+      STARTS_WITH_IC(Country.NAME, countryCode)
+    ));
+
+    if ( country != null ) {
+      normalizedCountry = country.getCode();
+    }
     
```

Comment:
why dont we do one find with country id = or iso code =

Reply:
do we need the name starts with?
if we get 2 chars as code that could return multiple countries
Cabo Verde
Cambodia
Cameroon
Canada
Cayman Islands, The

Reply:
+1 .. no need for separate queries anymore

Reply:
@adamfox-nanopay  .. the utility wasn't meant to be generic for all countries when it was written .. it was meant for local US and Canadian address normalization when they are returned from Flinks. This would need to be reworked if we need to worry about addresses in Cambodia or Cameroon.

Reply:
fair point but US is not going to return any useful country name either
country code or iso code should be sufficient no?

Reply:
@nanoscott We will keep that in mind so that we know where to look when we expand